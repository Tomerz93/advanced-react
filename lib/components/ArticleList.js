import React, {PureComponent} from 'react';
import Article from 'components/Article';

class ArticleList extends PureComponent {
  render() {
    let {articles} = this.props;
    return (
      <div>
        {
          Object.values(articles).map(article =>
            <Article
              key={article.id}
              article={article}
            />
          )

        }
      </div>
    );
  }
}

export default ArticleList;