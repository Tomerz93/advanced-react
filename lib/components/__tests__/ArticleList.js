import React from 'react';
import renderer from 'react-test-renderer';
import ArticleList from 'components/ArticleList';

describe('ArticleList', () => {
  it('should render correctly', () => {

    const testProps = {
      articles: {
        a: {id: 'a', date: 'Wed Jul 13 2016', title: 'a title', author: {firstName: 'a', lastName: 'a'}, body: 'a'},
        b: {id: 'b', date: 'Wed Jul 13 2016', title: 'b title', author: {firstName: 'a', lastName: 'a'}, body: 'a'}
      },
      store: {
        lookupAuthor: jest.fn(() => ({}))
      }
    };

    const tree = renderer.create(
      <ArticleList {...testProps} />
    ).toJSON();

    expect(tree).toMatchSnapshot();
    expect(tree.children.length).toBe(2);
  });
});
