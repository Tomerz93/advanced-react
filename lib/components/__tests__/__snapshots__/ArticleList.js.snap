// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList should render correctly 1`] = `
<div>
  <div
    style={
      Object {
        "borderBottomColor": "#aaa",
        "borderBottomStyle": "solid",
        "borderBottomWidth": 1,
        "marginBottom": 10,
        "paddingBottom": 10,
      }
    }
  >
    <div
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    >
      a title
    </div>
    <div
      style={
        Object {
          "fontSize": ".85rem",
        }
      }
    >
      Wed Jul 13 2016
    </div>
    <div
      style={
        Object {
          "paddingBottom": 10,
          "paddingTop": 10,
        }
      }
    >
      <a>
         
      </a>
    </div>
    <div
      style={
        Object {
          "paddingLeft": 20,
        }
      }
    >
      a
    </div>
  </div>
  <div
    style={
      Object {
        "borderBottomColor": "#aaa",
        "borderBottomStyle": "solid",
        "borderBottomWidth": 1,
        "marginBottom": 10,
        "paddingBottom": 10,
      }
    }
  >
    <div
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    >
      b title
    </div>
    <div
      style={
        Object {
          "fontSize": ".85rem",
        }
      }
    >
      Wed Jul 13 2016
    </div>
    <div
      style={
        Object {
          "paddingBottom": 10,
          "paddingTop": 10,
        }
      }
    >
      <a>
         
      </a>
    </div>
    <div
      style={
        Object {
          "paddingLeft": 20,
        }
      }
    >
      a
    </div>
  </div>
</div>
`;
